Lms/fix admin email verification

anathomical opened the pull request:
## WHAT
- Make sure that we flag users who sign up as admins with a Google or Clever account get marked as email-verified.
- Add a rake task to backfill the 14-ish admins who have signed up with Clever or Google and haven't been marked as verified
## WHY
While we don't want users signing up with Google or Clever to go through the email verification workflow, we do still want them to be marked as verified in our system for consistency.
## HOW
Add some logic the the account completion logic to just mark users as having verified their email when they're signing up with Clever or Google. 

### Notion Card Links
https://www.notion.so/quill/When-admins-sign-up-with-Google-or-Clever-set-their-Email-verification-status-to-Verified-ce876e661cb3406e906de936cd80d357?pvs=4

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Yes
Have you deployed to Staging? | Deploying now
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | N/A

anathomical committed:
Set email verification when creating new Clever/Google admins

anathomical committed:
Add a rake task to backfill email verification for Google/Clever admins

anathomical committed:
Try fixing linting error

anathomical committed:
Disable linting rule for one line

anathomical committed:
More linting stuff

emilia-friedberg commented on services/QuillLMS/lib/tasks/users.rake:
```diff
@@ -39,4 +39,19 @@ namespace :users do
+      .or(User.where.not(google_id: nil)) # note that the position of `or` matters a lot when you have multiple `where` clauses
+      .where(role: User::ADMIN)
+      # I was going to use the open-ended `..` here, but Rubocop apparently can't handle a `..` without a value at the end and chokes.  `..nil` is documented as being equivalent to an unterminated `..`
+      .where(created_at: DateTime.new(2023, 2, 27)..nil) # this is the date we launched self-service admin sign-up
```
I think this probably works, but there could theoretically be some admins that were invited by an existing admin (bypassing the admin flow and therefore email verification) who signed up in that timeframe, so it might be safer to check for the existence of an `approval_status` value to be sure that they really did use the self-service flow.

anathomical replied:
Good call.  Per our Zoom conversation, I'll update the logic here.

emilia-friedberg reviewed (approved):
One note on the rake task backfill, but I think it's theoretical anyway. Thank you for the quick fix!

anathomical committed:
Use AdminInfo instead of date_created to find admins to backfill

The presence of AdminInfo is a more precise way to determine if a user self-signed up as an admin than date of creation.

anathomical committed:
More linting fixes
